stdlib/_socket.pyi

```diff
@@ -650,8 +644,8 @@ SocketType = socket
 
 # ----- Functions -----
 
-def close(__fd: _FD) -> None: ...
-def dup(__fd: _FD) -> int: ...
+def close(__fd: SupportsIndex) -> None: ...
```

Comment:
We also have a `_typeshed.FileDescriptor` alias but it's just `int`. No change needed in this PR, but might be worth looking into whether we can make that alias usable here.